api/controllers/WebActionsController.js

```diff
@@ -238,11 +238,14 @@ module.exports = {
         // If available, attach the referralCode to the redirect URL
         if (referralCode.length > 0) {
           redirectUrl += `&referralCode=${referralCode}`;
-
+          // Get partner/campaign name to identify where a user signed up
+          let partnerOrCampaign = req.body.partner
```

Comment:
👀  nested ternary ops

If you really wanna try to keep this logic as concise as possible, I'd prefer we go with something like this instead:
```
{
  ...
  partner: req.body.partner || req.body.campaign
```

That should also take care of setting it to `undefined` if neither are set.